### ISSUE TITLE
`MessageSourceAutoConfigration` should handle dot-based package locations

### ISSUE THREAD

**Issue by mvitz** [spring-projects/spring-boot#10092]
The javadoc (and therefore the generated documentation on the property page) states that `basename` of `MessageSourceAutoConfiguration` should be set to

> Comma-separated list of basenames, each following the ResourceBundle convention.
> Essentially a fully-qualified classpath location. If it doesn't contain a package
> qualifier (such as "org.mypackage"), it will be resolved from the classpath root.

But when using the ResourceBundle convention by setting `spring.messages.basename=org.springframework.security.messages` the Condition specified in `MessageSourceAutoConfiguration` evaluates to false:

```
   MessageSourceAutoConfiguration:
      Did not match:
         - ResourceBundle did not find bundle with basename org.springframework.security.messages (MessageSourceAutoConfiguration.ResourceBundleCondition)
```

If I specify
`spring.messages.basename=org/springframework/security/messages` everything works as expected but this isn't the ResurceBundle convention.

**Comment by wilkinsona** [comment]
I don't think we can just correct the documentation as, assuming I've interpreted the code correctly, it won't work if you use a class rather than a .properties file for you resource bundle. And we can't just change the condition in 1.5.x as it will break anyone using the, arguably incorrect, slash-separated base name. We might be able to get away with correcting the condition and automatically replacing `/` with `.` in the configured base names.

**Comment by wilkinsona** [comment]
@snicoll has a nice idea for throwing a particular type of exception for an invalid configuration property value that includes its origin. In this case we could throw one when the basename uses `/` instead of `.` separates to help to guide users to the expected format.

**Comment by snicoll** [comment]
That's #10794

**Comment by wilkinsona** [comment]
Moving to RC1 so that the fix can build upon #10794.

**Comment by snicoll** [comment]
The auto-configuraton currently doesn't support class-based bundles.
